### ISSUE TITLE
Example doesn't work on 0.4.0

### ISSUE THREAD

**Issue by simgt** [zakarumych/egui-snarl#26]
Hey, thanks for the cool crate! Might put it to good use. 🙂

Commit `5df33348456c0a0e609c85073e38a6c280c3423e` seems to have broken the example, at least on macOS 14.5. Here's the error with the relevant part of the stack trace:

```
 RUST_BACKTRACE=1 cargo run -F serde -F egui-probe --example demo
    Updating crates.io index
   Compiling egui-snarl v0.4.0 (/Users/simon/Dev/egui-snarl)
    Finished dev [unoptimized + debuginfo] target(s) in 2.63s
     Running `target/debug/examples/demo`
thread 'main' panicked at /Users/simon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/egui-0.28.1/src/layout.rs:420:9:
assertion failed: max_rect.is_finite()
...
  11: egui_snarl::ui::<impl egui_snarl::Snarl<T>>::draw_node
             at ./src/ui.rs:1104:17
  12: egui_snarl::ui::<impl egui_snarl::Snarl<T>>::show::{{closure}}
             at ./src/ui.rs:606:32
  13: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/ops/function.rs:250:5
  14: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/alloc/src/boxed.rs:2015:9
  15: egui::containers::frame::Frame::show_dyn
             at /Users/simon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/egui-0.28.1/src/containers/frame.rs:280:19
  16: egui::containers::frame::Frame::show
             at /Users/simon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/egui-0.28.1/src/containers/frame.rs:270:9
  17: egui_snarl::ui::<impl egui_snarl::Snarl<T>>::show
             at ./src/ui.rs:546:9
  18: <demo::DemoApp as eframe::epi::App>::update::{{closure}}
             at ./examples/demo.rs:1006:13
...
```

Code works fine on `010664736db7e0998aa12288496cc811e9143be2` however.

**Comment by zakarumych** [comment]
Thank you for pointing to this.
I wasn't aware of the new `debug_assert` added and tested on release for some reason.

**Comment by zakarumych** [comment]
Fixed in da3ffb3f0a32736362e5daeef064c5863a064cb9

**Comment by zakarumych** [comment]
Fixed in da3ffb3f0a32736362e5daeef064c5863a064cb9
If happens again, reopen.
If you have any other issues, don't hesitate to open an issue :)

**Comment by simgt** [comment]
Thanks @zakarumych!
